app/models/concerns/teacher.rb

```diff
@@ -139,4 +141,54 @@ def update_teacher params
     end
     response
   end
-end
\ No newline at end of file
+
+  def is_premium?
+    subscriptions
+      .where("subscriptions.expiration >= ?", Date.today)
+      .where("subscriptions.account_limit >= ?", self.my_students.length)
+      .any?
+  end
+
+  def my_students
+    User.joins("JOIN classrooms ON users.classcode = classrooms.code")
+        .joins("JOIN users AS teachers ON teachers.id = classrooms.teacher_id")
+        .where("teachers.id = ?", self.id)
+  end
+
+  def is_trial_expired?
+    acss = self.teachers_activity_sessions_since_date(TRIAL_START_DATE)
+    acss.count > TRIAL_LIMIT
+  end
+
+  def teachers_activity_sessions_since_date date
+    ActivitySession.where(user: self.my_students)
+                   .where("completed_at >= ?", date)
+  end
+
+  def trial_activities_display_ratio
```

Comment:
'trial_activities_display_ratio' seems like a display helper method, so it should go in helpers/ probably either for users, teachers, subscriptions, or whatever.